lib/src/contracts/contract_invoker.dart

```diff
@@ -0,0 +1,118 @@
+part of 'package:web3dart/contracts.dart';
+
+class ContractInvocation {
+  final Web3Client client;
+  final DeployedContract contract;
+  final ContractFunction function;
+  List _params;
+
+  ContractInvocation(this.client, this.contract, this.function);
```

Comment:
I think this constructor should be private as well, so that a `ContractInvocation` can only be created with `DeployedContract.invoker`.
```suggestion
  ContractInvocation._(this._client, this._contract, this._function);
```